File: Kids-With-the-Greatest-Number-of-Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int max = *std::max_element(candies.begin(), candies.end());
        for (int candy : candies) {
            ans.push_back(candy + extraCandies >= max);
        }
        return ans;
    }
};

File: Kids-With-the-Greatest-Number-of-Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let ans = [];
    let max = Math.max(...candies);
    for(const candy of candies) {
        ans.push(candy + extraCandies >= max);
    }
    return ans;
};